generate_latest_json.py

```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env python3
+"""
+Sopel update-check JSON generation utility
+
+This script creates a `latest.json` file representing the most recent stable and
+unstable `sopel` releases published to PyPI, using PyPI's own JSON API.
+
+Copyright 2024 dgw, technobabbl.es
+Licensed under the Eiffel Forum License 2.
+
+https://sopel.chat
+"""
+import json
+
+from packaging.version import parse as parse_version
+import requests
+
+
+def main(argv=None):
+    data = requests.get('https://pypi.org/pypi/sopel/json').json()
+    releases = sorted([parse_version(v) for v in data['releases'].keys()])
+
+    latest = releases[-1]
+    stable = latest
+    if latest.is_prerelease:
+        for v in reversed(releases[:-1]):
+            if not v.is_prerelease:
+                stable = v
+                break
+        else:
+            print("ERROR: Latest version is a prerelease, but no earlier stable version found!")
+            exit(1)
```

Comment:
Nitpick: use `sys.exit(1)` or `raise SystemExit(1)` instead of `exit()` which [isn't guaranteed to be defined](https://docs.python.org/3/library/constants.html#constants-added-by-the-site-module)